api/volume.go

```diff
@@ -127,6 +129,18 @@ func (s *Server) VolumeCreate(rw http.ResponseWriter, req *http.Request) error {
 		}
 	}
 
+	if volume.AccessMode == "" && volume.FromBackup != "" {
+		_, volumeName, _, err := backupstore.DecodeBackupURL(volume.FromBackup)
+		if err != nil {
+			return errors.Wrapf(err, "failed to decode backup url: %v", volume.FromBackup)
+		}
+		bs, err := s.m.ListBackupsForVolumeSorted(volumeName)
+		if err != nil || bs == nil {
+			return errors.Wrapf(err, "failed to list backups for volumes: %v", volumeName)
+		}
+		volume.AccessMode = longhorn.AccessMode(bs[0].Status.Labels[types.LonghornVolumeAccessModeLabel])
```

Comment:
Is it possible that `bs[0]` does not have this label, whereas other have it?

Reply:
Since I will use `GetBackupRO` to get volume name from the backup, so it will be a object structure rather than list.

https://github.com/longhorn/longhorn-manager/blob/0d4593a5d1c666a1d1c8f6afeadffd2bde4a3c9f/webhook/resources/volume/mutator.go#L106-L115